### DISCUSSION TITLE
Can toktx read stdin?

### DISCUSSION THREAD

**Discussion by feugy** [KhronosGroup/KTX-Software#425] [Q&A] [answered]
Hello!

I'm crafting a one-liner to run imagemagick and toktx together.
Like other ktx command line tools, `toktx --help` mention reading from stdin.

However: `convert image.png -flop -strip - | toktx --uastc 4 image.ktx2` ends up in
> toktx: need some input files.

Could it be because of: https://github.com/KhronosGroup/KTX-Software/blob/877bca30f3c6eaca28628033180406f07cc34663/tools/toktx/toktx.cc#L698

Thanks for you help.

**Comment by MarkCallow** [comment, marked as answer]
> Could it be because of: 

Yes. It's deliberate because it may be necessary to rewind the stream while determining the file type and you can't rewind stdin. Also for many cases you need more than one input file.

I'm in the early stage of working on a stream implementation that will buffer and thus allow rewind. Leave this open to remind me to use that in `toktx` when it is complete. I started it for another purpose.
